src/widgets/programs/programs-detailed-summary.component.tsx

```diff
@@ -56,41 +55,46 @@ export default function ProgramsDetailedSummary(
           </thead>
           <tbody>
             {enrolledPrograms &&
-              enrolledPrograms.map(program => {
-                return (
-                  <React.Fragment key={program.uuid}>
-                    <tr
-                      className={`${
-                        program.dateCompleted
-                          ? `${styles.inactive}`
-                          : `${styles.active}`
-                      }`}
-                    >
-                      <td className="omrs-medium">{program.display}</td>
-                      <td>{dayjs(program.dateEnrolled).format("MMM-YYYY")}</td>
-                      <td>
-                        {program.dateCompleted
-                          ? `Completed on ${dayjs(program.dateCompleted).format(
-                              "DD-MMM-YYYY"
-                            )}`
-                          : "Active"}
-                      </td>
-                      <td>
-                        {
-                          <Link to={`${match.path}/${program.uuid}`}>
-                            <svg
-                              className="omrs-icon"
-                              fill="var(--omrs-color-ink-low-contrast)"
-                            >
-                              <use xlinkHref="#omrs-icon-chevron-right" />
-                            </svg>
-                          </Link>
-                        }
-                      </td>
-                    </tr>
-                  </React.Fragment>
-                );
-              })}
+              enrolledPrograms
+                .slice()
+                .sort((a, b) => (b.dateEnrolled > a.dateEnrolled ? 1 : -1))
```

Comment:
Should we perform all these changes already when setting the `enrolledPrograms`? Otherwise, we need to recompute every time we render. Alternatively, use `useMemo` to get a memoized version of it, with the memoization key being set to `enrolledPrograms`.

Reply:
Probably not. Perhaps it would be more appropriate to move these concerns to `programs.resource` so that `fetchEnrolledPrograms` would always return a sorted collection:

export function fetchEnrolledPrograms(
  patientID: string
): Observable<PatientProgram[]> {
  return openmrsObservableFetch(
    `/ws/rest/v1/programenrollment?patient=${patientID}`
  ).pipe(
    map(
      ({ data }) =>
        data["results"].sort((a, b) =>
          b.dateEnrolled > a.dateEnrolled ? 1 : -1
        ),
      take(3)
    )
  );
}

Reply:
Sounds good to me!

Reply:
Thanks @FlorianRappl!